featuregate/flag.go

```diff
@@ -5,40 +5,88 @@ package featuregate // import "go.opentelemetry.io/collector/featuregate"
 
 import (
 	"flag"
+	"fmt"
 	"strings"
 
 	"go.uber.org/multierr"
 )
 
 // NewFlag returns a flag.Value that directly applies feature gate statuses to a Registry.
+//
+// Deprecated: use NewFlags instead to get both gates and strict flags.
 func NewFlag(reg *Registry) flag.Value {
-	return &flagValue{reg: reg}
+	return &gatesFlagValue{args: &gateRegistrationArgs{reg: reg}}
 }
 
-// flagValue implements the flag.Value interface and directly applies feature gate statuses to a Registry.
-type flagValue struct {
-	reg *Registry
+// NewFlags returns two flag.Values: one that registers gates, and one that sets strict mode.
+func NewFlags(reg *Registry) (flag.Value, flag.Value) {
```

Comment:
If we ever want to introduce more flags with this approach we would have to deprecate the `NewFlags` function and add a new one that has one more value. What about returning a struct? e.g.

```go
type Flags struct {
    RegistrationFlag flag.Value
    StrictMode flag.Value
}
```

If this PR is still unclear, I would suggest we make just this change and then this PR does not block 1.0 readiness

Reply:
I filed #8727 to implement this and decouple the 1.0 release from this change